buildSrc/src/main/java/org/elasticsearch/gradle/ElasticsearchJavaPlugin.java

```diff
@@ -0,0 +1,378 @@
+            } else {
+                test.systemProperty("tests.seed", BuildParams.getTestSeed());
+            }
+
+            // don't track these as inputs since they contain absolute paths and break cache relocatability
+            File gradleHome = project.getGradle().getGradleUserHomeDir();
+            String gradleVersion = project.getGradle().getGradleVersion();
+            nonInputProperties.systemProperty("gradle.dist.lib", new File(project.getGradle().getGradleHomeDir(), "lib"));
+            nonInputProperties.systemProperty(
+                "gradle.worker.jar",
+                gradleHome + "/caches/" + gradleVersion + "/workerMain/gradle-worker.jar"
+            );
+            nonInputProperties.systemProperty("gradle.user.home", gradleHome);
+            // we use 'temp' relative to CWD since this is per JVM and tests are forbidden from writing to CWD
+            nonInputProperties.systemProperty("java.io.tmpdir", test.getWorkingDir().toPath().resolve("temp"));
+
+            nonInputProperties.systemProperty("compiler.java", BuildParams.getCompilerJavaVersion().getMajorVersion());
+            nonInputProperties.systemProperty("runtime.java", BuildParams.getRuntimeJavaVersion().getMajorVersion());
+
+            // TODO: remove setting logging level via system property
+            test.systemProperty("tests.logger.level", "WARN");
+            System.getProperties().entrySet().forEach(entry -> {
+                if ((entry.getKey().toString().startsWith("tests.") || entry.getKey().toString().startsWith("es."))) {
+                    test.systemProperty(entry.getKey().toString(), entry.getValue());
+                }
+            });
+
+            // TODO: remove this once ctx isn't added to update script params in 7.0
+            test.systemProperty("es.scripting.update.ctx_in_params", "false");
+
+            // TODO: remove this property in 8.0
+            test.systemProperty("es.search.rewrite_sort", "true");
+
+            // TODO: remove this once cname is prepended to transport.publish_address by default in 8.0
+            test.systemProperty("es.transport.cname_in_publish_address", "true");
+
+            // Set netty system properties to the properties we configure in jvm.options
+            test.systemProperty("io.netty.noUnsafe", "true");
+            test.systemProperty("io.netty.noKeySetOptimization", "true");
+            test.systemProperty("io.netty.recycler.maxCapacityPerThread", "0");
+
+            test.testLogging(logging -> {
+                logging.setShowExceptions(true);
+                logging.setShowCauses(true);
+                logging.setExceptionFormat("full");
+            });
+
+            if (OS.current().equals(OS.WINDOWS) && System.getProperty("tests.timeoutSuite") == null) {
+                // override the suite timeout to 30 mins for windows, because it has the most inefficient filesystem known to man
+                test.systemProperty("tests.timeoutSuite", "1800000!");
```

Comment:
This seems to be in the old impl too, but what's with the `!` suffix?

Reply:
From our own contributing docs, it seems the exclamation is important, although I have not found documentation on how it works. I'd like to leave this as a followup.